studymanager/src/main/java/io/redlink/more/studymanager/service/ElasticService.java

```diff
@@ -368,4 +371,32 @@ private Map<String, Object> toData(Map<String, Object> source) {
                 .forEach(k -> result.put(k.substring(5), source.get(k)));
         return result;
     }
+
+    private static <R, T extends Exception> R handleIndexNotFoundException(T e, Supplier<R> supplier) throws T {
+        return ElasticService.handleIndexNotFoundException(e, supplier, Function.identity());
+    }
+
+    private static <R, E extends Exception, T extends Exception> R handleIndexNotFoundException(E e, Supplier<R> supplier, Function<E, T> exceptionTFunction) throws T {
+        if (isElasticIndexNotFound(e)) return supplier.get();
+        throw exceptionTFunction.apply(e);
+    }
+
+    private static boolean isElasticIndexNotFound(Exception e) {
+        if (e instanceof ElasticsearchException ee) {
```

Comment:
Why are there two different if conditions? You can combine these into one with &&.

Reply:
True, but I find this way better readable.